backEnd/index.js

```diff
@@ -42,114 +16,106 @@ app.listen(port, () => {
   console.log(`Server listening on the port::${port}`);
 });
 
-
 //below are the API methods, matched with apiServices in the Front End
 //NOTE: requires firebase integration, uses "users" variable for testing
 
 // Get all users for searching. Client will filter the array that is returned
 // sends list of users up to the client
-app.get('/api/users', (req, res) => {
-  console.log("get all users")
-  //todo: Get users from Firebase
-  res.json(users);
+app.get('/api/users', async (req, res) => {
+  try{
+    console.log("get all users")
+    let users = await userRepository.getAllUsers();
+    res.json(users);
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
+    res.json(error);
+  }
 });
 
 // Gets a user by id - id is "path parameter"
 // sends a single user up to the client
-app.get("/api/users/:id", (req, res) => {
+app.get("/api/users/:id", async (req,res) => {
   let id = req.params.id;
-  console.log("Get user by ID: /users/:" + id);
-  //todo: Search user by ID in firebase and return that
-  res.json(users[0]);
+  try {
+    userValidator.validateId(id);
+
+    console.log("Get user by ID: /users/:" + id);
+    let found = await userRepository.getUserById(id);
+    console.log(found);
+    res.json(found);
+  }
+  catch(error) {
+    console.log('error' + error);
+    res.status(400);
```

Comment:
404 if the requested user doesn't exist. 400 if the request validation fails. 500 if the server fails